crates/wasmtime/src/instance.rs

```diff
@@ -498,15 +489,27 @@ impl<'a> ImportsBuilder<'a> {
             Extern::Memory(i) => {
                 self.memories.push(i.vmimport());
             }
+            Extern::Instance(i) => {
+                debug_assert!(Store::same(i.store(), self.matcher.store));
+                self.instances.push(unsafe { (*i.handle).clone() });
+            }
+            Extern::Module(m) => {
+                self.modules.push(m.clone());
+            }
         }
     }
 
-    fn imports(&self) -> Imports<'_> {
+    fn imports(&mut self) -> Imports<'_> {
```

Comment:
🚲shed: perhaps this method should be renamed to `build` now given the mutation and "builder" name?